### ISSUE TITLE
`AccessModifierIndentation` and `CommentIndentation` does not work well together

### ISSUE THREAD

**Issue by yskkin** [rubocop/rubocop#11237]
`# rubocop: enable` and `# rubocop:disable` may wrap whole method definition.
In such case, indentation should be based on wrapped method.

--------

## Expected behavior

No offenses.

## Actual behavior

Having one offense.
We cannot suppress offense as long as comment is followed by access modifier.

```
rubocop -c test.rubocop.yml foo.rb
Inspecting 1 file
C

Offenses:

foo.rb:5:3: C: [Correctable] Layout/CommentIndentation: Incorrect indentation detected (column 2 instead of 0).
  # rubocop:enable
  ^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

## Steps to reproduce the problem

Lint following file with the setting below:

foo.rb
```ruby
class A
  # rubocop:disable
  def foo
  end
  # rubocop:enable

private

  def bar
  end
end
```

test.rubocop.yml
```yml
AllCops:
  DisabledByDefault: true

Layout/AccessModifierIndentation:
  Enabled: true
  EnforcedStyle: outdent

Layout/CommentIndentation:
  Enabled: true
```

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
1.39.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 2.7.6) [x86_64-darwin21]
  - rubocop-rails 2.15.1
  - rubocop-rspec 2.11.1
```

**Comment by soroktree** [comment]
if there is another comment before #rubocop:enable, then what should be the alignment

`class A
  # rubocop:disable
  def foo
  end
  #some comment 
  # rubocop:enable

private

  def bar
  end
end`
